Tests/SentryTests/Integrations/URLSessionTaskMock.m

```diff
@@ -0,0 +1,24 @@
+#import "URLSessionTaskMock.h"
+
+@implementation URLSessionTaskMock {
```

Comment:
`l`: Can't we write that code in Swift?

Reply:
I tried, but I couldn't suppress the "deprecated" error of the init, this class was not meant to be override.

Reply:
Had the same issue. A comment to explain this would be nice. Something like this could work
https://github.com/getsentry/sentry-cocoa/blob/4063d291b19a6fa6f7b5190c30a19d6c1f495155/Tests/SentryTests/SentryCrash/TestSentryCrashAdapter.h#L7-L10